src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -22,42 +22,27 @@ internal abstract class AbstractConvertConcatenationToInterpolatedStringRefactor
     {
         public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
         {
-            // Currently only supported if there is no selection.  We could consider relaxing
-            // this if the selection is of a string concatenation expression.
-            if (context.Span.Length > 0)
-            {
-                return;
-            }
-
             var cancellationToken = context.CancellationToken;
 
             var document = context.Document;
             var position = context.Span.Start;
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             var token = root.FindToken(position);
 
-            // Cursor has to at least be touching a string token.
             var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            if (!token.Span.IntersectsWith(position) ||
-                !syntaxFacts.IsStringLiteral(token))
-            {
-                return;
-            }
-
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
 
             // The string literal has to at least be contained in a concatenation of some form.
             // i.e.  "goo" + a      or     a + "goo".  However, those concats could be in larger
             // concats as well.  Walk to the top of that entire chain.
-
             var literalExpression = token.Parent;
```

Comment:
nit: this now can be something other than a literal expression, maybe change the variable name to to `selectedExpression`. Similarly for the comment above, e.g. change "The string literal" to "The selected token"

Reply:
Thanks for the suggestion, I have made this change.
I also rebased on master again.